lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -1009,33 +1011,42 @@ sub select_coa {
             die $request->{_locale}->text('Invalid request');
         }
 
-        for my $coa_type (qw( chart gifi sic )) {
+        for my $coa_type (qw( chart sic )) {
             if ($request->{$coa_type}) {
                 if (! grep { $_ eq $request->{$coa_type} }
                     @{$coa_data->{$coa_lc}->{$coa_type}}) {
                     die $request->{_locale}->text('Invalid request');
                 }
             }
         }
-    }
 
-    if ($request->{coa_lc}){
-        if ($request->{chart}){
+        if ($request->{chart}) {
             my ($reauth, $database) = _get_database($request);
             return $reauth if $reauth;
 
-            $database->load_coa(
+            my $c = LedgerSMB::Company->new(
```

Comment:
Shouldn't instantiation of a new company happen only once and outside any conditionals?

Reply:
As discussed in the chat; yes, there's an entire rewrite of setup. Ok desired. This would be part of that rewrite. At the moment though, the company doesn't provide access to all functions that setup requires. So at this point, it doesn't make sense yet.